docs/source/hooks.rst

```diff
@@ -0,0 +1,175 @@
+3. :py:meth:`~flow.FlowProject.operation_hooks.on_success` will execute when the operation exits without an exception.
+4. :py:meth:`~flow.FlowProject.operation_hooks.on_exception` will execute when the operation exits with an exception.
+
+Hooks can be installed at the :ref:`operation level <operation-hooks>` or at the :ref:`FlowProject level <project-level-hooks>`.
+FlowProject-level hooks are called for every operation in the FlowProject.
+
+The hooks created with triggers :py:meth:`~flow.FlowProject.operation_hooks.on_start`, :py:meth:`~flow.FlowProject.operation_hooks.on_exit`, and :py:meth:`~flow.FlowProject.operation_hooks.on_success` require two arguments: the operation name and the :py:class:`signac.contrib.job.Job` object.
+Hooks created to trigger :py:meth:`~flow.FlowProject.operation_hooks.on_exception` require three arguments: the operation name, the output error, and the job object.
+
+.. note::
+
+    Hooks are run in the Python process where ``FlowProject.main()`` is called.
+    For this reason, hooks will not have access to modules in a container specified in the :term:`executable directive <executable>`.
+
+.. _operation-hooks:
+
+Operation Hooks
+===============
+
+Hooks may be added to individual operations using decorators.
+The :py:class:`~flow.FlowProject.operation_hooks` decorator tells **signac-flow** to run a hook (or set of hooks) when an operation reaches the specified trigger.
+
+The :py:class:`~flow.FlowProject.operation_hooks` decorator accepts objects as a function of the job operation (:py:class:`~flow.project.JobOperation`).
+
+An operation hook can be used to store basic information about the execution of a job operation in the job document.
+In the following example, when the test operation ``error_on_a_0`` raises an exception, the hook function ``store_error_to_doc`` executes.
+Otherwise, ``store_success_to_doc`` executes.
+
+.. code-block:: python
+
+    # project.py
+    from flow import FlowProject
+
+    class Project(FlowProject):
+        pass
+
+    def store_success_to_doc(operation_name, job):
+        job.doc.update({f'{operation_name}_success': True})
+
+    def store_error_to_doc(operation_name, error, job):
+        job.doc.update({f'{operation_name}_success': False})
+
+    @Project.operation
+    @Project.operation_hooks.on_success(store_success_to_doc)
+    @Project.operation_hooks.on_exception(store_error_to_doc)
+    @Project.post.isfile("result.txt")
+    def error_on_a_0(job):
+        if job.sp.a == 0:
+            # Have jobs with statepoint 'a' == 0 fail
+            raise RuntimeError("Cannot process jobs with a == 0.")
+
+    if __name__ == '__main__':
+        Project().main()
+
+
+If ``error_on_a_0`` is executed using ``python project.py run --operation error_on_a_0 --filter a 1``, the ``on_success`` hook trigger will run, and ``job.doc.error_on_a_0_success`` will be ``True``.
```

Comment:
It may be helpful to remind the reader that `error_on_a_0` is an operation name. At least, I had to refresh myself by reading the code again to see what that name was.
```suggestion
If the operation ``error_on_a_0`` is executed on jobs with state point key ``a`` equal to 1 using ``python project.py run --operation error_on_a_0 --filter a 1``, the ``on_success`` hook trigger will run, and ``job.doc.error_on_a_0_success`` will be ``True``.
```